staging/src/k8s.io/apiserver/pkg/endpoints/handlers/namer.go

```diff
@@ -100,15 +100,6 @@ func (n ContextBasedNaming) GenerateLink(req *http.Request, obj runtime.Object)
 		return "", fmt.Errorf("missing requestInfo")
 	}
 
-	if len(namespace) == 0 && len(name) == 0 {
-		if len(requestInfo.Name) == 0 {
```

Comment:
Seems it's better to check `len(namespace) == 0` in `ObjectName`. And here delete the outer `if len(namespace) == 0 && len(name) == 0` but keep the inner `if len(requestInfo.Name) == 0`.